p_set_1.rb

```diff
@@ -2,17 +2,48 @@
 # SMART WATER
 BOTTLE_VOLUME = 16
 WATER_FLOW_RATE = 192
+
 def bottle_reporter
-  puts "That's #{bottle_calculator get_minutes} bottles of water"
+  puts "That's #{bottle_calculator ask_minutes} bottles of water"
 end
 
 def bottle_calculator(minutes)
   minutes * WATER_FLOW_RATE / BOTTLE_VOLUME
 end
 
-def get_minutes
-  puts "How many minutes do you shower for?"
+def ask_minutes
+  puts "How many mins do you shower for?"
   minutes = gets.chomp.to_i
 end
 
+# ITSA MARIO
+SPACE =      " "
+BRICKS =     "#"
+APEX_WIDTH = 2
+
+def ask_height
+  while true
+    puts "height?"
+    height = gets.chomp.to_i
+    break if height < 24
+  end
+  height
+end
+
+def calculate_row_length(height)
+  row_length = height + 1
+end
+
+def pyramid_printer(row_length)
+  n = APEX_WIDTH
+  for n in n..row_length
+    puts SPACE * (row_length - n) + BRICKS * n
+  end
+end
+
+def mario
+  pyramid_printer calculate_row_length ask_height
```

Comment:
I found this hard to read. I'd much prefer it spread over multiple lines to be more readable.
```
height = ask_height
row_length = calculate_row_length(height)
pyramid_printer(row_length)
```